packages/core/src/types.ts

```diff
@@ -189,18 +189,19 @@ export interface UserOperationRequest_v7 {
 
 //#region UserOperationRequest
 // Reference: https://eips.ethereum.org/EIPS/eip-4337#definitions
-export type UserOperationRequest<TEntryPointVersion extends EntryPointVersion> =
-  TEntryPointVersion extends "0.6.0"
-    ? UserOperationRequest_v6
-    : TEntryPointVersion extends "0.7.0"
-    ? UserOperationRequest_v7
-    : never;
+export type UserOperationRequest<
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
+> = TEntryPointVersion extends "0.6.0"
+  ? UserOperationRequest_v6
+  : TEntryPointVersion extends "0.7.0"
+  ? UserOperationRequest_v7
+  : never;
```

Comment:
we should change this never to union type no?

https://github.com/alchemyplatform/aa-sdk/pull/607/files#diff-887ed5859d7652b1d3596c31911d16babe9c65fe8b94ae1c92762907291791d7R73 would resolve as never no?

Reply:
ditto: as above.

Reply:
By the way, I thought I'd point out that another way to write these "type-level switch statements" which is a little shorter is
```ts
export type UserOperationRequest<
  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
> = {
  "0.6.0": UserOperationRequest_v6;
  "0.7.0": UserOperationRequest_v7;
}[TEntryPointVersion];
```
but the current way is fine too.

Reply:
oh neato @dphilipson ! yea I like that format a lot